### ISSUE TITLE
make_axes_2d KeyError

### ISSUE THREAD

**Issue by yallup** [handley-lab/anesthetic#321]
Using anesthetic from pypi, anesthetic 2.1.0

```python

from anesthetic import make_2d_axes
#this works fine
make_2d_axes([0,1])

#this throws a KeyError
make_2d_axes([1,2])

```

`make_2d_axes` errors unless 0 is included in the index seemingly. Default behavior of anesthetic is to interact with dataframes through int column labels.

```
  File "pandas/_libs/index.pyx", line 147, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 176, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 2606, in pandas._libs.hashtable.Int64HashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 2630, in pandas._libs.hashtable.Int64HashTable.get_item
KeyError: 0

```

Edit: appears to break if any integer in the sequence is missing, 
```python
make_2d_axes([0,2])
```
also breaks

**Comment by williamjameshandley** [comment]
We've had issues with integer labels before #163 #304 #215.

The code should work if you use strings, e.g. 'x0', 'x1', or even '0', '1'. pandas prefers this to mixed object indexes for a variety of reasons but I agree this is not ideal behaviour and is a bug. If you build a failing test from the above examples in a new PR I can have a look at the best way to fix it.

**Comment by appetrosyan** [comment]
This is preventing the update of paranestamol to work with anesthetic 2.0. So I'd be happy to help with the PR

**Comment by yallup** [comment]
Solved by #322 , thanks @williamjameshandley !
